app/feature/feature-forever/src/main/kotlin/com/hedvig/android/feature/forever/ui/ForeverContent.kt

```diff
@@ -0,0 +1,165 @@
+import androidx.compose.foundation.layout.windowInsetsPadding
+import androidx.compose.foundation.layout.windowInsetsTopHeight
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.pullrefresh.PullRefreshDefaults
+import androidx.compose.material.pullrefresh.PullRefreshIndicator
+import androidx.compose.material.pullrefresh.pullRefresh
+import androidx.compose.material.pullrefresh.rememberPullRefreshState
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.IconButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.core.designsystem.component.button.HedvigContainedButton
+import com.hedvig.android.core.designsystem.component.button.HedvigTextButton
+import com.hedvig.android.core.ui.appbar.m3.TopAppBarLayoutForActions
+import com.hedvig.android.feature.forever.ForeverUiState
+import com.hedvig.android.feature.forever.data.toErrorMessage
+import hedvig.resources.R
+import javax.money.MonetaryAmount
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+internal fun ForeverContent(
+  uiState: ForeverUiState,
+  reload: () -> Unit,
+  onShareCodeClick: (code: String, incentive: MonetaryAmount) -> Unit,
+  onCodeChanged: (String) -> Unit,
+  onSubmitCode: (String) -> Unit,
+  openReferralsInformation: (String, MonetaryAmount) -> Unit,
+) {
+  val systemBarInsetTopDp = with(LocalDensity.current) {
+    WindowInsets.systemBars.getTop(this).toDp()
+  }
+  val pullRefreshState = rememberPullRefreshState(
+    refreshing = uiState.isLoading,
+    onRefresh = reload,
+    refreshingOffset = PullRefreshDefaults.RefreshingOffset + systemBarInsetTopDp,
+  )
+
+  val sheetState = rememberModalBottomSheetState(true)
+  val coroutineScope = rememberCoroutineScope()
+  var showEditBottomSheet by rememberSaveable { mutableStateOf(false) }
+
+  LaunchedEffect(uiState.showEditCode) {
+    coroutineScope.launch {
+      if (uiState.showEditCode) {
+        sheetState.expand()
+      } else {
+        sheetState.hide()
+      }
+      showEditBottomSheet = uiState.showEditCode
+    }
+  }
```

Comment:
Doesn't this make it so that when the sheet comes in it just pops in instead of animating in? Since if `showEditCode` is true, it will first try to animate the state (at that point `showEditBottomSheet` is still false, so no UI is showing) and then it sets it to true, so the bottom sheet shows, already in the expanded state?
That's what I see from reading the code at least.

That's why I thought this is a bit tricky to do with the UI state determining the sheet state while also having to have a local representation of it too. If I am wrong ignore this. If I am right we can look into this later I think.

Reply:
Yes that is true. Lets figure this out together!